File: Programming Basics/Conditional Statements Advanced - Exercises/04.FishingBoat.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        double budget = double.Parse(Console.ReadLine());
        string season = Console.ReadLine();
        int fisherCount = int.Parse(Console.ReadLine());

        double shipRentCost = 0;

        switch (season)
        {
            case "Summer":
            case "Autumn":
                shipRentCost = 4200;
                break;
            case "Spring":
                shipRentCost = 3000;
                break;
            case "Winter":
                shipRentCost = 2600;
                break;
        }
        if (fisherCount <= 6)
        {
            shipRentCost *= 0.90;
        }
        else if (fisherCount >= 7 && fisherCount <= 11)
        {
            shipRentCost *= 0.85;
        }
        else if (fisherCount > 12)
        {
            shipRentCost *= 0.75;
        }
        if (fisherCount % 2 == 0 && season != "Autumn")
        {
             shipRentCost *= 0.95; 
        }
        if (budget >= shipRentCost)
        {
            Console.WriteLine($"Yes! You have {budget - shipRentCost:f2} leva left.");
        }
        else
        {
            Console.WriteLine($"Not enough money! You need {shipRentCost - budget:f2} leva.");
        }
    }
}

File: Programming Basics - Python/Conditional Statements Advanced - Exercises/04.FishingBoat.py
budget = float(input())
season = input()
fishermen = int(input())

rent = 0
discount = 0

if season == 'Spring':
    rent = 3000
elif season == 'Summer' or season == 'Autumn':
    rent = 4200
elif season == 'Winter':
    rent = 2600

if fishermen <= 6:
    discount = 0.10
elif 7 <= fishermen <= 11:
    discount = 0.15
elif fishermen >= 12:
    discount = 0.25

total = rent - (rent * discount)

if season != 'Autumn' and fishermen % 2 == 0:
    total *= 0.95

if budget >= total:
    print(f'Yes! You have {budget - total:.2f} leva left.')
else:
    print(f'Not enough money! You need {total - budget:.2f} leva.')